sql/src/main/java/io/crate/executor/task/SetSessionTask.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.executor.task;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.core.collections.Row;
+import io.crate.executor.JobTask;
+import io.crate.metadata.settings.session.SessionSettingApplier;
+import io.crate.metadata.settings.session.SessionSettingRegistry;
+import io.crate.operation.projectors.RepeatHandle;
+import io.crate.operation.projectors.RowReceiver;
+import io.crate.sql.tree.Expression;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class SetSessionTask extends JobTask {
+
+    private static final ESLogger LOGGER = Loggers.getLogger(SetSessionTask.class);
+
+    private final SessionContext sessionContext;
+    private final Map<String, List<Expression>> settings;
+
+    public SetSessionTask(UUID jobId, Map<String, List<Expression>> settings, SessionContext sessionContext) {
+        super(jobId);
+        this.sessionContext = sessionContext;
+        this.settings = settings;
+    }
+
+    @Override
+    public void execute(RowReceiver rowReceiver, Row parameters) {
+        for (Map.Entry<String, List<Expression>> setting : settings.entrySet()) {
+            SessionSettingApplier applier = SessionSettingRegistry.getApplier(setting.getKey());
+            if (applier != null) {
+                applier.apply(parameters, setting.getValue(), sessionContext);
```

Comment:
@dobe `set` allows only string, bool, numericexpressions and ident, ON, DEFAULT, all of those can be simply converted to strings. Something like `cast` will fail much earlier with the parser exception. I can't really find a way to make it fail in SetSessionTask, except of creating an expression, which is not allowed for `set` by the parser, in unit tests for this task. Shall i do smth like that, or it would be fine just to enclose the logic in execute method of `setsessiontask` in try/catch?